builder/comp-builder.nix

```diff
@@ -356,6 +356,7 @@ let
       description = package.synopsis or "";
       license = haskellLib.cabalToNixpkgsLicense package.license;
       platforms = if platforms == null then lib.platforms.all else platforms;
+      mainProgram = if haskellLib.isExecutableType componentId then exeName else "";
```

Comment:
Nice!  `exeName` includes the `extension` (`.exe` on windows for instance).  It does not look like [getExe](https://github.com/NixOS/nixpkgs/blob/8ef7523c8e11da7fd23e6b87371e7aa1eab718bb/lib/meta.nix#L141-L142) adds it so I guess including it is the right thing to do.

Can you please update so that it does not set `mainProgram` to `""`.  Something like:
```
  } // lib.optionalAttrs (haskellLib.isExecutableType componentId) {
    mainProgram = exeName;
  };
```

Reply:
Got it, I'll change it so it doesn't set `mainProgram` for non-executable components. Thanks!